### PR TITLE
Addon-docs: Add DocsPage automagically

### PR DESCRIPTION
**Pull request by shilman**
Issue: #7316

## What I did

- Made framework-specific docs presets smarter to add `docs: DocsPage` param
- Remove this from examples

This is backwards compatible because users can still set `docs` param to whatever they want in `config.js`.

## How to test

Test the example storybooks (Angular, HTML, Vue, Official, CRA)

### CODE REVIEW

**Review by igor-dv — APPROVED** [review]
Nice 🙂

File: addons/docs/common/makePreset.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+const common = require('./preset');
+
+module.exports = framework => {
```
Comment (ndelangen) [inline addons/docs/common/makePreset.js]: So this preset creates a preset on the fly and refers to the file it generated?

File: addons/docs/common/makePreset.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const { addParameters } = require('@storybook/${framework}');
+const { DocsPage } = require('@storybook/addon-docs/blocks');
+
+addParameters({
+  docs: DocsPage,
+});
+  `.trim();
+  fs.writeFileSync(`${configModule}.js`, configContents);
+
+  function config(entry = []) {
```
Comment (ndelangen) [inline addons/docs/common/makePreset.js]: Is this an existing preset property?

**Comment by ndelangen** [comment]
Quite confusing, but I guess it works :)

**Comment by shilman** [comment]
The other option is to create a separate file for each framework, unless you know of a way to add a global parameter independent of the framework

File: addons/docs/common/makePreset.js (hunk 2 above)
Comment (shilman) [reply addons/docs/common/makePreset.js]: Yeah

File: addons/docs/common/makePreset.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+const common = require('./preset');
+
+module.exports = framework => {
+  const configModule = `${__dirname}/config-${framework}`;
+  const configContents = `
+const { addParameters } = require('@storybook/${framework}');
```
Comment (tmeasday) [inline addons/docs/common/makePreset.js]: Could we instead make it possible to import `addParameters` from `@storybook/core` or something to avoid this?

**Review by tmeasday — COMMENTED** [review]
I'm not really sure the writing a file on the fly thing is a good idea. Seems likely to cause trouble. Also probably inefficient to do it every time we start storybook.

I've wanted to be able to import `addParameter` from other places many times before. Should we just bite the bullet and export it already?

**Pull request merged**